go/internal/initutil/node.go

```diff
@@ -328,18 +338,33 @@ func (m *Manager) getGRPCServer() (*grpc.Server, *grpcgw.ServeMux, error) {
 		grpc_zap.ReplaceGrpcLoggerV2(grpcLogger)
 		grpcLoggerConfigured = true
 	}
+
+	// noop auth func
+	authFunc := func(ctx context.Context) (context.Context, error) { return ctx, nil }
+
+	if m.Node.Protocol.needAuth {
```

Comment:
maybe you can just skip this boolean check to check that secret and pk are not empty? this will avoid updaitng the main.go file to set "needAuth" to true